services/compute/virtualmachine/virtualmachine.go

```diff
@@ -144,14 +144,38 @@ func (c *client) getWssdVirtualMachineStorageConfigurationImageReference(s *comp
 	return *s.Name, nil
 }
 func (c *client) getWssdVirtualMachineStorageConfigurationOsDisk(s *compute.OSDisk) (*wssdcloudcompute.Disk, error) {
-	if s.Vhd == nil {
-		return nil, errors.Wrapf(errors.InvalidInput, "Vhd Configuration is missing in OSDisk")
+	if s.ManagedDisk == nil {
+		if s.Vhd == nil {
+			return nil, errors.Wrapf(errors.InvalidInput, "Vhd Configuration is missing in OSDisk")
+		}
+		if s.Vhd.URI == nil {
+			return nil, errors.Wrapf(errors.InvalidInput, "Vhd URI Configuration is missing in OSDisk")
+		}
 	}
-	if s.Vhd.URI == nil {
-		return nil, errors.Wrapf(errors.InvalidInput, "Vhd URI Configuration is missing in OSDisk")
+
+	diskName := ""
+	if s.Vhd != nil && s.Vhd.URI != nil {
+		diskName = *s.Vhd.URI
+	}
```

Comment:
we should fail if any of them are `nil`, right?

Reply:
if ManagedDisk is set, then these can be nil, since we're using ManagedDisk to set the SecurityProfile, but we still want to be able to use a gallery image (so we would set only ManagedDisk and leave vhd as nil)